### PR TITLE
Python: update config-generation to use new package path

### LINKED ISSUE #1502: Configgen: update Python package name
### What

Change the configgen to produce Python package names of the form `...{api}_{version}.gapic`

### Why

All configs in googleapis/googleapis use this newer form, but the generator produces the older form `{api}.{version}.gapic`

### PR DESCRIPTION
**Pull request by landrito**
Resolves #1502

### CODE REVIEW

**Comment by landrito** [comment]
Added _ as a character in the point release matcher. This will be used for speech 1.1 point release.

File: src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java [hunk 1]
```diff
@@ -161,4 +161,21 @@ public String rewrite(String input) {
       return input.replaceAll(pattern, replacement);
     }
   }
+
+  private static class PythonLanguageFormatter implements LanguageFormatter {
+    private static final RewriteRule PYTHON_REWRITE_RULE =
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java]: This duplicates the `commonRewriteRules`. How about passing it in as a constructor parameter?

File: src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java [hunk 2]
```diff
@@ -161,4 +161,21 @@ public String rewrite(String input) {
...
+        new RewriteRule("^google(?!\\.cloud)", "google.cloud");
+
+    @Override
+    public String getFormattedPackageName(String packageName) {
+      packageName = PYTHON_REWRITE_RULE.rewrite(packageName);
+      List<String> names = Arrays.asList(packageName.split("\\."));
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java]: Replate with `Splitter.on(DEFAULT_PACKAGE_SEPARATOR).split(packageName)`

File: src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java [hunk 3]
```diff
@@ -161,4 +161,21 @@ public String rewrite(String input) {
...
+      if (VersionMatcher.isVersion(names.get(names.size() - 1))) {
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java]: Just trying to understand what this does: what's an example where this check would return true?

File: src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java (hunk 1 above)
Comment (landrito) [reply src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java]: Done.

File: src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java (hunk 2 above)
Comment (landrito) [reply src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java]: Done.

File: src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java (hunk 3 above)
Comment (landrito) [reply src/main/java/com/google/api/codegen/configgen/transformer/LanguageTransformer.java]: This handles situations where the package name does not end with a major version such as with longrunning. I added a longrunning test to make sure this case is covered.

**Pull request merged**